### PR TITLE
Set-AzureVMChefExtension command in ARM

### PR DESCRIPTION
**Pull request by NimishaS**
In Progress

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@NimishaS__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by markcowl** [comment]
@azuresdkci retest this please

**Comment by markcowl** [comment]
@NimishaS Thanks for the contribution, were you going to add online help for this cmdlet?

**Comment by NimishaS** [comment]
@markcowl , this is still in testing phase. I will confirm once I am done. I am going to update the help files also.

**Comment by stankovski** [comment]
@NimishaS is this PR ready to be merged?

**Comment by NimishaS** [comment]
@stankovski, @markcowl  we are facing issues while testing it. While running the command `Set-AzureVMChefExtension`, we get this error:
```
Set-AzureVMChefExtension : Run Login-AzureRmAccount to login.
At line:1 char:1
+ Set-AzureVMChefExtension -ResourceGroupName "testresourcegrp1" -VMName "test-vm- ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Set-AzureVMChefExtension], PSInvalidOperationException
    + FullyQualifiedErrorId : InvalidOperation,Microsoft.Azure.Commands.Compute.Extension.Chef.SetAzureVMChefExtension
```
We get the same error even after running `Login-AzureRmAccount` command. It doesn't go inside our code at all.
Can you please suggest something here?

**Comment by stankovski** [comment]
@huangpf can you take a look at the cmdlet to make sure the base cmdlet is prperly executed?

File: src/ResourceManager/Compute/Commands.Compute/Common/ConstantStringTypes.cs [hunk 1]
```diff
@@ -87,6 +87,7 @@ public static class ProfileNouns
         public const string VirtualMachineExtensionImage = "AzureRmVMExtensionImage";
         public const string VirtualMachineExtensionImageVersion = "AzureRmVMExtensionImageVersion";
         public const string VirtualMachineExtensionImageType = "AzureRmVMExtensionImageType";
+        public const string VirtualMachineChefExtension = "AzureVMChefExtension";
```
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/Common/ConstantStringTypes.cs]: ```AzureRmVMChefExtension```

**Comment by huangpf** [comment]
@NimishaS 
Try to rename your cmdlet to ```*AzureRm*```, so that it's not overlapped with the RDFE ones. Thanks.

**Comment by NimishaS** [comment]
@huangpf I tried with `AzureRmVMChefExtension` too,  but still facing the same issue. While debugging I found that `DefaultProfile.Context` is null here https://github.com/Azure/azure-powershell/blob/dev/src/ResourceManager/Common/Commands.ResourceManager.Common/AzureRMCmdlet.cs#L75, which results into the error that we are facing.
I have tried commands like `Select-AzureRmProfile`, `Set-AzureRmContext` etc, but nothing helped.

One more thing, I face the same issue when I try to run other commands like `Set-AzureRmVmBGInfoExtension` too. It seems like I am missing something while testing the code in Debug mode. I am novice to `C#`.

**Comment by huangpf** [comment]
How did you setup the debug? Here is the instruction: https://github.com/Azure/azure-powershell/wiki/Microsoft-Azure-PowerShell-Developer-Guide#importing-modules-automatically-when-start-debugging

**Comment by NimishaS** [comment]
@huangpf we are following the instructions from https://github.com/Azure/azure-powershell/wiki/Microsoft-Azure-PowerShell-Developer-Guide#importing-modules-automatically-when-start-debugging only. Below is the value of `Command line arguments` that we are using:
```
-NoExit -Command "C:\Users\azure\nim\azure-powershell\src\Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Compute\Microsoft.Azure.Commands.Compute.dll"
```

**Comment by huangpf** [comment]
@hyonholee @stankovski 
Any thoughts on why the wiki about debug settings doens't work as expected for @NimishaS?

**Comment by markcowl** [comment]
@NimishaS This isn't actually following the advice - it needs to use ```Import-Module Package\Debug\ResourceManager\AzureResourceManager\AzureRM.Compute\AzureRm.Compute.psd1```

**Comment by markcowl** [comment]
@NimishaS Is this PR ready to merge?

**Comment by NimishaS** [comment]
@markcowl , thanks for help. Importing `AzureRm.Compute.psd1` helped but we had to modify the paths of some modules inside `AzureRm.Compute.psd1` to get this working.
The command is executing successfully for both `windows` and `linux`, but it's currently not picking the default value of `-TypeHandlerVersion`. We need to fix that and also check if there is any other such option.
We'll update this very soon.

**Comment by NimishaS** [comment]
@markcowl , we have tested this. It can be merged now.

**Pull request merged**